include/oneapi/dpl/functional

```diff
@@ -17,7 +17,18 @@
 #define _ONEDPL_FUNCTIONAL
 
 #include "oneapi/dpl/internal/common_config.h"
+
+// Silence the deprecation warnings of the negators injected from <functional>
+// The macro definition is contained in this header to avoid redefinition errors on the user side
```

Comment:
In my understanding - please correct if I am wrong - the "silence" macro effectively does not add the deprecation attribute to the negation functions. That means that a) if the standard header has been included before oneDPL headers, the warnings will still be there, and b) if we include the header with the "silence" macro defined, it will impact the whole translation unit anyway.

The direct conclusion from (b) is that undefining the macro makes no good whatsoever, and maybe even harms, by disabling the warnings and leaving no clue it has been made.

Reply:
Overall, I think we should not do this suppression. If we position our headers as substitutes for the standard ones (i.e., these can be used instead), in most if not all cases we should not change the behavior of the standard headers.